BasiliskII/src/Unix/main_unix.cpp

```diff
@@ -436,6 +439,17 @@ int main(int argc, char **argv)
 		} else if (strcmp(argv[i], "--rominfo") == 0) {
 			argv[i] = NULL;
 			PrintROMInfo = true;
+		} else if (strcmp(argv[i], "--switch") == 0) {
+			argv[i] = NULL;
+			if (argv[++i] == NULL) {
+				printf("switch address not defined\n");
+				usage(argv[0]);
+			}
+			int vde_str_len = strlen(argv[i]);
+			vde_sock = (char*)malloc(vde_str_len);
+			memset(vde_sock,NULL,vde_str_len);
+			strncpy(vde_sock,argv[i],vde_str_len);
```

Comment:
I think lines 448 to 451 can be replaced with strdup(argv[i]).

Although, do we really need to copy it? If we're setting to NULL in the array, why not just point to the original content?

Reply:
Ping

Reply:
I'm not following. You are passing pointers to the vde_open() API...

Reply:
again it was my mistake. Yes, I am passing pointers to vde_open, you were right, I have changed the code, there is no need for any malloc or copy, just a pointer passing will do.